src/main/java/org/yaml/snakeyaml/scanner/ScannerImpl.java

```diff
@@ -230,9 +230,24 @@ public ScannerImpl(StreamReader reader, LoaderOptions options) {
     fetchStreamStart();// Add the STREAM-START token.
   }
 
+  /**
+   * Check whether the next token is the given type.
+   */
+  @Override
+  public boolean checkToken(Token.ID choice) {
+    while (needMoreTokens()) {
```

Comment:
I wonder how to measure that the code duplication provides the performance improvement.

Reply:
I will try to write up a JMH benchmark and tests to quantify the before/after difference for a variety of inputs.

In ad-hoc local testing, parsing very large GiB+ yaml files via `org.yaml.snakeyaml.Yaml#parse` this change yields ~25% speedup (from > 4 minutes before to < 3 minutes after).

Reply:
I think this might be a major boost as there are only 11 references to the varargs version and 53 with 1 argument. It meant in all those 53 invocations the `new ID { arg }` is created :) 

I think this can be done in Java 7 sources without `default` method.
This will of course will break things for those who use Scanner interface.

But I am not sure how many actually provide Custom Scanners (even in test it is used only once - CanonicalScanner).

Or we maybe can merge this after this release. As the next one is planned to move to Java 8.

@asomov What do you think ?

Reply:
I am inclining to the next version inclusion (because of that backward compatibility issue and close release date), even I would like to see this improvement ASAP :)

Reply:
Yes, I think it would be better for the next version (2.4)
Version 2.3 is almost there: https://bitbucket.org/snakeyaml/snakeyaml/wiki/Changes